src/praisonai/praisonai/integrations/compute/docker.py

```
@@ -392,28 +392,53 @@ async def shutdown(self, instance_id: str) -> None:
         await loop.run_in_executor(None, self._shutdown_sync, instance_id)
 
+    def _find_container(self, instance_id: str):
+        """Look up a container by name, for instances this process didn't start.
+
+        ``_containers`` only knows about instances provisioned by *this*
+        process, so a CLI invocation (``praisonai managed stop``) would
+        otherwise find nothing and silently no-op.
+        """
+        try:
+            return self._get_client().containers.get(f"praisonai_{instance_id}")
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)